src/mechanics/mesh/MeshGmsh.cpp

```diff
@@ -0,0 +1,378 @@
+    if (line.compare("$ENDELEMENTS") != 0)
+        throw NuTo::Exception(__PRETTY_FUNCTION__, "$EndElements not found!");
+}
+
+
+void ReadPhysicalNamesASCII(std::istream& file, GmshFileContent& fileContent)
+{
+    std::string line;
+    std::getline(file, line);
+    int numNames = std::stoi(line);
+
+    fileContent.physicalNames.resize(numNames);
+    for (GmshPhysicalNames& name : fileContent.physicalNames)
+    {
+        file >> name.dimension;
+        file >> name.id;
+        file >> name.name;
+
+        // remove quotation marks from string
+        name.name.erase(std::remove(name.name.begin(), name.name.end(), '\"'), name.name.end());
+        std::getline(file, line);
+    }
+    std::getline(file, line);
+
+    std::transform(line.begin(), line.end(), line.begin(), ::toupper);
+    if (line.compare("$ENDPHYSICALNAMES") != 0)
+        throw NuTo::Exception(__PRETTY_FUNCTION__, "$EndPhysicalNames not found!");
+}
+
+void ProcessSectionASCII(std::istream& file, GmshFileContent& fileContent)
+{
+    std::string line;
+    std::getline(file, line);
+    if (line[0] == '$')
+    {
+        // remove $
+        std::string sectionName = line.substr(1, line.size() - 1);
+        // capitalize
+        std::transform(sectionName.begin(), sectionName.end(), sectionName.begin(), ::toupper);
+
+        // Execute section read function
+        if (sectionName.compare("NODES") == 0)
+            ReadNodesASCII(file, fileContent);
+        else if (sectionName.compare("ELEMENTS") == 0)
+            ReadElementsASCII(file, fileContent);
+        else if (sectionName.compare("PHYSICALNAMES") == 0)
+            ReadPhysicalNamesASCII(file, fileContent);
+        else
+            throw NuTo::Exception(__PRETTY_FUNCTION__, "Unhandled gmsh section type: " + sectionName);
+    }
+}
+
+
+const NuTo::InterpolationSimple& CreateElementInterpolation(NuTo::MeshFem& mesh, int gmshType, int dimension)
+{
+    using namespace NuTo;
+    switch (gmshType)
+    {
+    case 1:
+        return mesh.CreateInterpolation(InterpolationTrussLinear(dimension));
+    case 2:
+        return mesh.CreateInterpolation(InterpolationTriangleLinear(dimension));
+    case 3:
+        return mesh.CreateInterpolation(InterpolationQuadLinear(dimension));
+    default:
+        throw NuTo::Exception(__PRETTY_FUNCTION__, "Unhandled gmsh element type.");
+    }
+}
+
+
+std::string GetPhysicalGroupName(const GmshFileContent& fileContent, int groupId)
+{
+    std::string physicalName{""};
+    for (const GmshPhysicalNames& gmshPhysicalName : fileContent.physicalNames)
+        if (gmshPhysicalName.id == groupId)
+        {
+            physicalName = gmshPhysicalName.name;
+            break;
+        }
+    return physicalName;
+}
+
+
+// Member functions
+// %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+NuTo::MeshGmsh::MeshGmsh(const std::string& fileName)
+{
+    ReadGmshFile(fileName);
+}
```

Comment:
Why not simply put the contents of `ReadGmshFile` here?

Reply:
Maybe we will provide additional ctors with additional options in the future. For example if you want just a single physical group of the gmsh file. But we can remove the extra function call if you want.